tests/manage/test_pvc_delete_verify_size_is_returned_to_backendpool.py

```diff
@@ -0,0 +1,257 @@
+    template_data['pvc_size'] = '10Gi'
+    create_oc_resource(
+        'pvc.yaml', TEMPLATES_DIR, _templating, template_data, "CSI/rbd"
+    )
+
+    # Validate pvc is in bound state
+    assert PVC.wait_for_resource(condition="Bound", resource_name=pvc_name)
+
+    # Validate pvc is created on ceph
+    pvc_info = PVC.get(resource_name=pvc_name)
+
+    name = get_ceph_tools_pod()
+    po = pod.Pod(name, namespace=defaults.ROOK_CLUSTER_NAMESPACE)
+    cmd = f"rbd ls -p {pool_name}"
+    pvc, _, _ = po.exec_command(cmd=cmd, timeout=20)
+
+    assert pvc_info['spec']['volumeName'] in pvc
+
+
+def create_pod(pod_name, pvc_name):
+    """
+    Create a app pod
+    """
+
+    template_data['pod_name'] = pod_name
+    template_data['pvc_name'] = pvc_name
+    create_oc_resource(
+        'pod.yaml', TEMPLATES_DIR, _templating, template_data, "CSI/rbd"
+    )
+
+    assert POD.wait_for_resource(condition="Running", resource_name=pod_name)
+
+
+def run_io(pod_name):
+    """
+    Run io on the mount point
+    """
+
+    run_cmd(
+        f"oc rsh -n {PROJECT_NAME} {pod_name}"
+        " dd if=/dev/urandom of=/var/lib/www/html/dd_ar bs=10M count=950"
+    )
+
+
+@pytest.fixture(scope='class')
+def test_fixture(request):
+    """
+    Fixture for the test
+    """
+
+    self = request.node.cls
+
+    def finalizer():
+        teardown(self)
+    request.addfinalizer(finalizer)
+    setup(self)
+
+
+def setup(self):
+    """
+    Setting up the environment for the test
+    """
+
+    create_cephblock_pool(self.pool_name)
+    create_storageclass(self.sc_name, self.pool_name)
+    create_secret(self.secret_name, self.pool_name)
+    PRJ.new_project(PROJECT_NAME)
+
+
+def teardown(self):
+    """
+    Tearing down the environment for the test
+    """
+    assert PRJ.delete(resource_name=PROJECT_NAME)
+    assert SECRET.delete(resource_name=self.secret_name)
+    assert SC.delete(resource_name=self.sc_name)
+    assert CBP.delete(resource_name=self.pool_name)
+
+
+@tier1
+@pytest.mark.usefixtures(
+    test_fixture.__name__,
+)
+class TestPVCDeleteAndVerifySizeIsReturnedToBackendPool(ManageTest):
+    """
+    Testing after pvc deletion the size is returned to backendpool
+    """
+    pool_name = "rbd-pool"
+    sc_name = "rbd-storageclass"
+    pvc_name = "rbd-pvc"
+    pod_name = "rbd-pod"
+    secret_name = "csi-rbd-secret"
+
+    def test_pvc_delete_and_verify_size_is_returned_to_backend_pool(self):
+        """
+        Test case to verify after delete pvc size returned to backend pools
+        """
+
+        used_before_creating_pvc = check_ceph_used_space()
+        create_pvc(self.pvc_name, self.sc_name, self.pool_name)
+        create_pod(self.pod_name, self.pvc_name)
+        run_io(self.pod_name)
+        used_after_creating_pvc = check_ceph_used_space()
+        assert used_before_creating_pvc < used_after_creating_pvc
+        assert POD.delete(resource_name=self.pod_name)
+        assert PVC.delete(resource_name=self.pvc_name)
```

Comment:
@Akarsha-rai we would need to confirm that on deleting a PVC, the underlying PV is deleted both from OC and the ceph backend.

Reply:
done